platform/app/src/hooks/useSearchParams.js

```diff
@@ -1,17 +1,23 @@
-import useQuery from './useQuery';
+import { useLocation } from 'react-router';
 
 /**
- * It returns a Map of the query parameters in the URL, where the keys are
- * lowercase
- * @returns A function that returns a Map of the query parameters.
+ * It returns a URLSearchParams of the query parameters in the URL, where the keys are
+ * either lowercase or maintain their case based on the lowerCaseKeys parameter.
+ * @param {boolean} lowerCaseKeys true to return lower case keys; false (default) to maintain casing;
+ * @returns {URLSearchParams}
  */
-export default function useSearchParams() {
-  const query = useQuery();
-  // make query params case-insensitive
-  const searchParams = new Map();
-  for (const [key, value] of query) {
-    searchParams.set(key.toLowerCase(), value);
+export default function useSearchParams(lowerCaseKeys = false) {
+  const searchParams = new URLSearchParams(useLocation().search);
+
+  if (!lowerCaseKeys) {
+    return searchParams;
+  }
+
+  const lowerCaseSearchParams = new URLSearchParams();
+
+  for (const [key, value] of searchParams) {
+    lowerCaseSearchParams.set(key.toLowerCase(), value);
   }
```

Comment:
this lower casing utility probably is usefull for dicom tags too, can we put it in utilties?